cargo-dist/src/platform.rs

```diff
@@ -0,0 +1,261 @@
+    TARGET_X86_WINDOWS,
+};
+use cargo_dist_schema::ArtifactId;
+
+use crate::{config::ZipStyle, DistGraphBuilder, ReleaseIdx, SortedMap, TargetTriple};
+
+/// Suffixes of TargetTriples that refer to statically linked linux libcs.
+///
+/// On Linux it's preferred to dynamically link libc *but* because the One True ABI
+/// is actually the Linux kernel syscall interface, you *can* theoretically statically
+/// link libc. This comes with various tradeoffs but the big selling point is that the
+/// Linux kernel is a much more slowly moving target, so you can build a binary
+/// that's portable across way more systems by statically linking libc. As such,
+/// for any archive claiming to provide a static libc linux build, we can mark this
+/// archive as providing support for any linux distro (for that architecture)
+///
+/// Currently rust takes "linux-musl" to mean "statically linked musl"a
+/// in the future it will mean "dynamically linked musl":
+///
+/// https://github.com/rust-lang/compiler-team/issues/422
+///
+/// We prefer "musl-static" and "musl-dynamic" aliases to disambiguate this situation,
+/// but support bare musl with the current semantics:
+///
+/// FIXME: when rustc makes the change, move "linux-musl" from STATIC_LIBCS to REPLACEABLE_LIBCS!
+/// (this may involve detecting the current rust toolchain).
+const LINUX_STATIC_LIBCS: &[&str] = &["linux-musl-static", "linux-musl"];
+/// Dynamically linked linux libcs that static libcs can replace
+const LINUX_STATIC_REPLACEABLE_LIBCS: &[&str] = &["linux-gnu", "linux-musl-dynamic"];
+/// A fake TargetTriple for apple's universal2 format (staples x64 and arm64 together)
+const TARGET_MACOS_UNIVERSAL2: &str = "universal2-apple-darwin";
+
+/// The quality of support an archive provides for a given platform
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum SupportQuality {
+    /// The archive natively supports this platform, there's no beating it
+    HostNative,
+    /// The archive natively supports this platform, but it's a Universal binary that contains
+    /// multiple platforms stapled together, so if there are also more precise archives, prefer those.
+    BulkyNative,
+    /// The archive is still technically native to this platform, but it's in some sense
+    /// imperfect. This can happen for things like "running a 32-bit binary on 64-bit" or
+    /// "using a statically linked linux libc". This solution is acceptable, but a HostNative
+    /// (or BulkyNative) solution should always be preferred.
+    ImperfectNative,
+    /// The archive is only running by the grace of pretty heavyweight emulation like Rosetta2.
+    /// This should be treated as a last resort, but hey, it works!
+    Emulated,
+    /// The layers of emulation are out of control.
+    Hellmulated,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum RuntimeCondition {
+    MinGlibcVersion { major: u64, series: u64 },
+    MinMuslVersion { major: u64, series: u64 },
```

Comment:
i never quite worked out how these would get updated to concrete values post-build. mostly these are sketches. I think maybe the idea would be that you'd rerun `PlatformSupport::new` post-builds and it would know to add these conditions with the refined info from the DistManifest or something (similar to how linkage works).

Reply:
at very least we could just default these to the values we hardcode installers now so the value is in one central location.